docker/idf/Dockerfile

```
@@ -13,8 +16,13 @@ USER developer
 ENV LC_ALL=C.UTF-8 LANG=C.UTF-8
 
 ARG idf_version
-RUN mkdir ~/git && git clone -b ${idf_version} --recursive https://github.com/espressif/esp-idf.git ~/git/esp-idf
-COPY idf_tools.py.patch git/esp-idf/tools/idf_tools.py.patch
-RUN cd ~/git/esp-idf && git apply ~/git/esp-idf/tools/idf_tools.py.patch
-RUN cd ~/git/esp-idf && ./install.sh
-RUN echo "source $HOME/git/esp-idf/export.sh" >> ~/.bashrc
+RUN sudo git clone -b ${idf_version} --recursive https://github.com/espressif/esp-idf.git /opt/esp-idf
+WORKDIR /opt/esp-idf
+COPY idf_tools.py.patch idf_tools.py.patch
+RUN sudo git apply idf_tools.py.patch
+ENV IDF_TOOLS_PATH=/opt/espressif
+RUN sudo -E python3 tools/idf_tools.py install
+RUN sudo -E python3 tools/idf_tools.py install-python-env
+RUN python3 tools/idf_tools.py export >> ~/.bashrc
```

You set links, so you could use them here as well. I guess this is intended for later compilation?

```suggestion
RUN sudo -E python tools/idf_tools.py install
RUN sudo -E python tools/idf_tools.py install-python-env
RUN python tools/idf_tools.py export >> ~/.bashrc
```

---

It's a simple solution to force idf to use py3